Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -0,0 +1,309 @@
+﻿using NationalInstruments.DataInfrastructure;
+using NationalInstruments.ModularInstruments;
+using NationalInstruments.ModularInstruments.NIRfsg;
+using NationalInstruments.ModularInstruments.SystemServices.TimingServices;
+using System;
+using System.Linq;
+
+namespace NationalInstruments.ReferenceDesignLibraries.Methods
+{
+    public static class EnvelopeTracking
+    {
+
+        #region Type Definitions
+        public struct EnvelopeGeneratorConfiguration
+        {
+            public RfsgTerminalConfiguration TerminalConfiguration;
+
+            public static EnvelopeGeneratorConfiguration GetDefault()
+            {
+                return new EnvelopeGeneratorConfiguration()
+                {
+                    TerminalConfiguration = RfsgTerminalConfiguration.Differential,
+                };
+            }
+        }
+
+        public struct TrackerConfiguration
+        {
+            public double InputImpedance_Ohms;
+            public double CommonModeOffset_V;
+            public double Gain_VperV;
+            public double OutputOffset_V;
+
+            public static TrackerConfiguration GetDefault()
+            {
+                return new TrackerConfiguration()
+                {
+                    InputImpedance_Ohms = 1e6,
+                    CommonModeOffset_V = 1.0,
+                    Gain_VperV = 2.5,
+                    OutputOffset_V = 2.55
+                };
+            }
+        }
+
+        public enum DetroughType { 
+            Exponential, 
+            Cosine, 
+            Power
+        };
+
+        public struct DetroughConfiguration
+        {
+            public DetroughType Type;
+            public double MinimumVoltage;
+            public double MaximumVoltage;

```

Comment:
Normally I would say these need units, but that might be a bit redundant because the name already includes them... thoughts?

Reply:
I guess adding a unit makes it consistent and also clarifies its in volts and not mV or something like that.

Reply:
b534e30f32480f976d7205ec2c4d5d8a0fd45c3b